cli/Cargo.toml

```diff
@@ -47,9 +47,6 @@ required-features = ["test-fakes"]
 [[test]]
 name = "runner"
 
-[build-dependencies]
-cargo_metadata = { workspace = true }
```

Comment:
Ah, thanks! Can probably go from the root `Cargo.toml` too

Reply:
Woops, removed from root as well now 👍

Reply:
note: I do have a script I wrote to catch unused dependencies since `ouroboros` slipped through the cracks at one point, but it's on one of my branches: https://github.com/martinvonz/jj/blob/push-qvrktsnwpkzu/tools/scripts/unused_workspace_deps.py

I wonder if I should submit it upstream as a separate check or something.

Reply:
Another option might be https://github.com/est31/cargo-udeps. Anyway, I'll resolve this comment now so this can be merged.